conmon/conmon.c

```diff
@@ -319,8 +321,30 @@ int write_k8s_log(int fd, stdpipe_t pipe, const char *buf, ssize_t buflen)
 			}
 		}
 
+		write_len = line_len;
+
+		/*
+		 * We need to remove \r from a \r\n sequence while writing to the file.
+		 * This can be taken out once kubernetes enables OPOST in the client.
+		 */
+		if (*line_end == '\n' && line_len > 1) {
+			line_end--;
+			if (*line_end == '\r') {
+				buf[line_len - 2] = '\n';
```

Comment:
I think the proper way is to just not send the \r or \n from the buffer, but append "\n" as a separate writev segment.
(Of course, that still leaves the question about CR-LF vs attached clients)